Tools/WinMLRunner/src/Run.cpp

```diff
@@ -178,6 +180,126 @@ HRESULT CreateSession(LearningModelSession& session, IDirect3DDevice& winrtDevic
                 WINML_PROFILING_STOP(profiler, WINML_MODEL_TEST_PERF::CREATE_SESSION);
             }
         }
+#ifdef MCDM_BUILD
+        else if ((TypeHelper::GetWinmlDeviceKind(deviceType) != LearningModelDeviceKind::Cpu) &&
+                 (!adapterName.empty() || adapterIndex != -1))
+        {
+             com_ptr<IDXCoreAdapterFactory> spFactory;
+             THROW_IF_FAILED(DXCoreCreateAdapterFactory(IID_PPV_ARGS(spFactory.put())));
+
+             com_ptr<IDXCoreAdapterList> spAdapterList;
+             const GUID dxGUIDs[] = { DXCORE_ADAPTER_ATTRIBUTE_D3D12_CORE_COMPUTE };
+
+             THROW_IF_FAILED(spFactory->GetAdapterList(dxGUIDs, ARRAYSIZE(dxGUIDs), spAdapterList.put()));
+
+             CHAR driverDescription[128];
```

Comment:
It would be better to dynamical determine the size needed using QueryPropertySize()

Reply:
Changed to dynamicly determine size needed